File: packages/javascript/lib/word/randJob.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * The trades and roles people hold. Never a person name — which is why there is no `Baker` or `Hunter` here.
 *
 * `randWord({ theme: 'job' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randJob({ language: 'ko', count: 3 }); // ['기사', '마법사', '대장장이']
 * randJob({ language: 'en', count: 3 }); // ['Wizard', 'Ranger', 'Blacksmith']
 */
export function randJob(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Job words with the language and theme each one came from.
 *
 * @example
 * randJob({ language: 'ko', output: 'detail' });
 * // [{ word: '기사', language: 'ko', theme: 'job' }]
 */
export function randJob(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randJob(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('job', options);
}

File: packages/dart/lib/src/word/rand_job.dart
import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// The trades and roles people hold. Never a person name — which is why there is no `Baker` or `Hunter` here.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.job] to `randWordDetails` — Dart has
/// no overloads, and twenty-four more functions for it would be twenty-four too many.
///
/// ```dart
/// randJob(language: WordLanguage.ko, count: 3); // [기사, 마법사, 대장장이]
/// randJob(language: WordLanguage.en, count: 3); // [Wizard, Ranger, Blacksmith]
/// ```
List<String> randJob({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,
}) => randWord(
  language: language,
  theme: WordTheme.job,
  count: count,
  realism: realism,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
);
